source/javascripts/pages/WorkflowsPage/components/WithBlockPanel/WithBlockPanel.tsx

```diff
@@ -0,0 +1,53 @@
+import { Box, Input, Notification, TagsInput, Text } from '@bitrise/bitkit';
+import useNavigation from '../../../PipelinesPage/hooks/useNavigation';
+import { Step } from '@/models';
+
+type WithBlockPanelProps = {
+  step: Step;
+};
+
+const WithBlockPanel = (props: WithBlockPanelProps) => {
+  const { step } = props;
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { image, services } = step.withBlockData!;
```

Comment:
I'd only pass the component the required params, and the the entire step object.